disable Upgrade and Save (UpSell) box for lower plan

ciprianimike opened the pull request:
#### Proposed Changes

* Temporarily comment out `Upgrade and Save` box for  plan upsell.
* Temporarily comment out `Upgrade and Save` box Tests for plan upsell.

#### Testing Instructions

1. Visit /start/
2. Select a free domain
3. Select starter plan ( free is not selectable on the plans page with the actual header)
4. From the checkout page remove the Starter plan from the cart
5. Verify the Upgrade and Save box doesn't show up

#### Screenshots
* Before
<img width="985" alt="Screenshot 2022-06-06 at 16 35 55" src="https://user-images.githubusercontent.com/5706607/172183261-0c75e97b-22fe-435c-a601-d1ccd2a97382.png">

* After
<img width="986" alt="Screenshot 2022-06-06 at 16 37 42" src="https://user-images.githubusercontent.com/5706607/172183279-5aadc56c-324f-4dab-af43-3e8b64cb6fa7.png">


#### Related Issue
* MarTech-708

ciprianimike committed:
disable Upgrade and Save (UpSell) box for lower plan

dzver commented on client/my-sites/checkout/composite-checkout/components/secondary-cart-promotions.tsx:
```diff
@@ -2,7 +2,13 @@ import config from '@automattic/calypso-config';
 import { FunctionComponent, useMemo } from 'react';
 import { useSelector } from 'react-redux';
-import CartFreeUserPlanUpsell from 'calypso/my-sites/checkout/cart/cart-free-user-plan-upsell';
+/**
```
Commenting out is not a preferred way to disable features. We can disable the feature conditionally by making sure the box is not there if one of the new plans is in the shopping cart.

ciprianimike committed:
Checkout / Upsell Box enable via config data

ciprianimike commented:
@dzver I changed approaches, using the json config file.

ciprianimike commented:
> @dzver I changed approach, using the json config file.
> 
> [There's a failed E2E test](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/plans/plans__legacy-renew.ts#L40) I'm investigating

The test was solved by re-running it. Ready for review

DavidRothstein reviewed (changes requested):
This looks like it would disable the upsell box in the `horizon` environment only, but leave it enabled in the other 5 environments (since you have `checkout_sellup_box` set to `true` there...)

Also, why would we want to disable this?  I think https://github.com/Automattic/wp-calypso/pull/64370 already takes care of switching it back to a Personal plan upsell, which seems like the correct behavior.

Commit:
add control on cart products

User isPro() and isStarter() to determine the plan product and eventually show the Plan Upsell

Commit:
reset Cart Upsell test to trunk
